lib/letsencrypt.js

```diff
@@ -36,180 +39,81 @@ function createServer(app, settings, certificates, leSettings, log) {
-                    });
-                } catch (err) {
-                    log.error('LetsEncrypt challenge server could not be started: ' + err);
+            const gl = require('greenlock').create(glOpts);
+            gl.manager.defaults({
+                agreeToTerms: true,
+                subscriberEmail: leSettings.email
+            }).then(() => {
+                // Add sites.
+                // TODO: something seems fishy with the timing of this because on the first run
+                // a warning about no sites being configured is found, even though things go on to
+                // work perfectly.
+                for (const domain of leSettings.domains) {
+                    log.debug(`Adding to Greenlock: ${domain}`);
+                    gl.add({ subject: domain, altnames: [domain] });
                 }
-            }
-
-            const options    = deepClone(certificates);
-            const defaultTls = tls.createSecureContext(certificates);
-            let hostTls;
-            let running;
-
-            options.SNICallback = function (hostname, cb) {
-                if (leSettings.domains.indexOf(hostname) !== -1) {
-                    if (settings.leUpdate) {
-                        if (running === true) {
-                            cb(null, hostTls || defaultTls);
-                        } else
-                        if (running) {
-                            running.push(cb);
-                        } else {
-                            running = [cb];
 
-                            return lex.httpsOptions.SNICallback(hostname, (err, tls) => {
-                                tls && log.debug('Got valid certificates from letsencrypt');
-                                err && log.error('Cannot get certificates: ' + err);
-                                lex.debug = false;
-                                hostTls = tls;
-                                for (let r = 0; r < running.length; r++) {
-                                    running[r](err, tls || defaultTls);
-                                }
-                                running = true;
+                require('greenlock-express')
+                    .init({
+                        greenlock: gl
+                    }).ready(glx => {
+                        // The below is from greenlock-express examples/https/server.js
+
+                        // Start the challenge server if configured
+                        if (settings.leUpdate) {
+                            const httpServer = glx.httpServer();
+                            settings.lePort = parseInt(settings.lePort, 10) || 80;
+                            const bind = (!settings.bind || settings.bind === '0.0.0.0') ? undefined : settings.bind || undefined;
+                            httpServer.listen(settings.lePort, bind, function () {
+                                console.info('Listening on ', httpServer.address());
```

Comment:
I would replace console.log with log.info

Reply:
Well spotted! Updated in https://github.com/ioBroker/ioBroker.js-controller/pull/1150/commits/b3a9819a23df0512d83bdd82cfadc1799d456af2